src/tools/generate_workspace/src/main/java/com/google/devtools/build/workspace/GenerateWorkspace.java

```diff
@@ -59,7 +60,13 @@ public static void main(String[] args) {
       return;
     }
 
-    GenerateWorkspace workspaceFileGenerator = new GenerateWorkspace(options.outputDir);
+    if ((options.buildOutputFile.isEmpty() && options.workspaceOutputFile.isEmpty())
+        != options.outputDir.isEmpty()) {
```

Comment:
With the current logic, `generate_workspace --workspace_file=whatever --output_dir=whatever` wouldn't throw an error (`options.buildOutputFile.isEmpty() && options.workspaceOutputFile.isEmpty()) != options.outputDir.isEmpty()` -> `(true && false) != false` -> `false != false`.